doc/user_guide.md

```diff
@@ -435,27 +434,49 @@ Refer to spdm_server_init() in [spdm_responder.c](https://github.com/DMTF/spdm-e
 
 3. Register message process callback
 
-   This callback need handle both SPDM vendor defined message and transport layer application message.
-   ```
-   return_status libspdm_get_response_vendor_defined_request (
-     void           *spdm_context,
-     const uint32_t *session_id,
-     bool            is_app_message,
-     size_t          request_size,
-     const void     *request,
-     size_t         *response_size,
-     void           *response
-   )
-   {
-     if (is_app_message) {
-       // this is a transport layer application message
-     } else {
-       // this is a SPDM vendor defined message (without transport layer header)
-     }
-   }
-
-   libspdm_register_get_response_func (spdm_context, libspdm_get_response_vendor_defined_request);
-   ```
+   3.1 This callback handles transport layer application message.
+    ```
+    return_status libspdm_get_response (
+      void           *spdm_context,
+      const uint32_t *session_id,
+      bool            is_app_message,
+      size_t          request_size,
+      const void     *request,
+      size_t         *response_size,
+      void           *response
+    )
+    {
+      if (is_app_message) {
+        // this is a transport layer application message
+      } else {
+        // other SPDM message
+      }
+    }
+
+    libspdm_register_get_response_func (spdm_context, libspdm_get_response);
+    ```
+    3.2 This callback handles SPDM Vendor Defined Commands
+    ```
+    libspdm_return_t libspdm_vendor_response_func(

```

Comment:
Do you want to return the header and payload, or payload only?

For former, it could be libspdm_get_response_vdm_message.
For latter, it could be libspdm_get_response_vdm_payload.

Reply:
Both header and payload I want returned.